pybamm/expression_tree/operations/evaluate_python.py

```diff
@@ -310,7 +310,7 @@ def find_symbols(
 
         elif isinstance(symbol, pybamm.SparseStack):
             if len(children_vars) == 1:
-                symbol_str = children_vars[0]
+                symbol_str = children_vars[0]  # pragma: no cover
```

Comment:
again a test would be good for this. if you don't want to do it in this PR (understandable) then an issue for reminder would be good too.

Reply:
more below which I won't mention. Looks like we were not covering a lot of it :(

Reply:
Quite a few of these were introduced as 'indirect' changes during one of the commits. I will go back and see whether these can be relaxed again, or if new tests are needed...

Reply:
I've gone through and removed all of the 'no cover' statements that I inserted due to indirect changes in previous revisions. Codecov seems to have settled itself down now and is passing everything again. So we should be good.